Sources/CCFilterNode/CCFilterNode.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import Foundation
+
+private protocol CCFilterNodeDelegate: class {
+    func childNodeDidChangeSelection(_ childNode: CCFilterNode)
+}
+
+public class CCFilterNode {
+
+    // MARK: - Public properties
+
+    public let title: String
+    public let name: String
+    public var value: String?
+    public var numberOfResults: Int
+
+    public var isSelected: Bool {
+        didSet { delegate?.childNodeDidChangeSelection(self) }
+    }
+
+    // MARK: - Private properties
+
+    private(set) var children: [CCFilterNode] = []
+    private weak var delegate: CCFilterNodeDelegate?
+
+    // MARK: - Setup
+
+    public init(title: String, name: String, value: String? = nil, isSelected: Bool = false, numberOfResults: Int = 0) {
+        self.title = title
+        self.name = name
+        self.value = value
+        self.isSelected = isSelected
+        self.numberOfResults = numberOfResults
+    }
+
+    // MARK: - Public methods
+
+    func add(child: CCFilterNode, at index: Int? = nil) {
+        if let index = index { children.insert(child, at: index) }
+        else { children.append(child) }
+        child.delegate = self
+    }
+
+    func child(at index: Int) -> CCFilterNode {
```

Comment:
Shouldn't it be optional in the index is out of bounds? You could also implement custom subscript instead of this function.

Reply:
I think the view controller should make sure the index is valid before calling this function. I simply use this function for consistency, I have the function to add a child so I think it is nice to have a function to get a child as well :)

Reply:
I think it's easy to forget this check in the view controller and get a crash.

Reply:
Yes, much better to assume the caller might use a bad index. We have the custom `[safe: index]` subscript that is always good to use